### PR TITLE
Move Streamly.FileSystem.Handle to streamly-core

### PR DESCRIPTION
**Pull request by rnjtranjan**

### CODE REVIEW

**Comment by rnjtranjan** [comment]
fixes:https://github.com/composewell/streamly/issues/1835

File: core/src/Streamly/Internal/FileSystem/Handle.hs [hunk 1]
```diff
@@ -213,12 +211,12 @@ getChunkOf = undefined
 -- read may be less than or equal to @size@.
 -- @since 0.9.0
 {-# INLINE _getChunksWith #-}
-_getChunksWith :: (IsStream t, MonadIO m)
-    => Int -> Handle -> t m (Array Word8)
-_getChunksWith size h = go
-  where
+_getChunksWith :: (MonadIO m)
+    => Int -> Handle -> S.Stream m (Array Word8)
```
Comment (harendra-kumar) [inline core/src/Streamly/Internal/FileSystem/Handle.hs:215]: Use the `Stream` type unqualified.

File: core/src/Streamly/Internal/FileSystem/Handle.hs [hunk 2]
```diff
@@ -420,6 +422,10 @@ putChunk h arr = A.asPtrUnsafe arr $ \ptr ->
 -- Writing
 -------------------------------------------------------------------------------
 
+{-# INLINE mapM'_ #-}
+mapM'_ :: Monad m => (a -> m b) -> Stream m a -> m ()
+mapM'_ f = D.mapM_ f . S.toStreamD
```
Comment (harendra-kumar) [inline core/src/Streamly/Internal/FileSystem/Handle.hs:427]: Use `drainBy`.

File: core/streamly-core.cabal [hunk 3]
```diff
@@ -331,6 +331,11 @@ library
                      , Streamly.Internal.Unicode.Char.Parser
                      , Streamly.Internal.Unicode.Array.Char
 
+                      -- Filesystem/IO
+                      , Streamly.FileSystem.Handle
+                      , Streamly.Internal.FileSystem.Handle
+
+
```
Comment (harendra-kumar) [inline core/streamly-core.cabal:338]: remove one blank line.

File: src/Streamly/Internal/FileSystem/File.hs [hunk 4]
```diff
@@ -98,20 +98,18 @@ import Streamly.Data.Fold (chunksOf)
 import Streamly.Internal.Control.Concurrent (MonadAsync)
 import Streamly.Internal.Data.Array.Unboxed.Type (Array(..), writeNUnsafe)
 import Streamly.Internal.Data.Fold.Type (Fold(..))
-import Streamly.Internal.Data.Stream.Type (Stream)
-import Streamly.Internal.Data.Stream.IsStream.Type (IsStream)
+import Streamly.Data.Stream (Stream)
```
Comment (harendra-kumar) [inline src/Streamly/Internal/FileSystem/File.hs:101]: The changes in this file are already made in another PR.

File: core/streamly-core.cabal [hunk 5]
```diff
@@ -331,6 +331,11 @@ library
                      , Streamly.Internal.Unicode.Char.Parser
                      , Streamly.Internal.Unicode.Array.Char
 
+                      -- Filesystem/IO
+                      , Streamly.FileSystem.Handle
```
Comment (harendra-kumar) [inline core/streamly-core.cabal:335]: This should go in the exposed modules section below.

File: core/src/Streamly/Internal/FileSystem/Handle.hs [hunk 6]
```diff
@@ -124,25 +124,23 @@ import Streamly.Internal.Data.Unfold.Type (Unfold(..))
...
 import qualified Streamly.Data.Fold as FL
 import qualified Streamly.Data.Array.Unboxed as A
 import qualified Streamly.Internal.Data.Array.Unboxed.Type as A
-    (byteLength, asPtrUnsafe)
-import qualified Streamly.Internal.Data.Array.Unboxed.Mut as MArray
-    (Array(..), newPinnedArrayBytes, asPtrUnsafe)
-import qualified Streamly.Internal.Data.Array.Stream.Foreign as AS
+import qualified Streamly.Internal.Data.Array.Unboxed.Mut.Type as MArray
+--import qualified Streamly.Internal.Data.Array.Unboxed as AS
+import qualified Streamly.Internal.Data.Array.Stream.Mut.Foreign as AS
```
Comment (harendra-kumar) [inline core/src/Streamly/Internal/FileSystem/Handle.hs:137]: Let's move `Streamly.Internal.Data.Array.Stream.Foreign` to `streamly-core` in a separate PR, and then base this PR on that one.

File: core/src/Streamly/Internal/FileSystem/Handle.hs [hunk 7]
```diff
@@ -356,19 +354,23 @@ readWith = UF.many A.read readChunksWith
 readWithBufferOf :: MonadIO m => Unfold m (Int, Handle) Word8
 readWithBufferOf = readWith
 
+{-# INLINE concat' #-}
+concat' :: (Monad m, Unboxed a) => S.Stream m (Array a) -> S.Stream m a
+concat' m = S.fromStreamD $ D.unfoldMany A.read (S.toStreamD m)
```
Comment (harendra-kumar) [inline core/src/Streamly/Internal/FileSystem/Handle.hs:359]: Let's move Array.Stream.Foreign to streamly-core and use it from there.

**Review by harendra-kumar — CHANGES_REQUESTED** [review]

**Pull request merged**